extension/src/schemas/vscode-notebook.ts

```diff
@@ -22,6 +28,15 @@ export const CellMetadata = Schema.Struct({
     key: Schema.String,
     value: Schema.Unknown,
   }).pipe(Schema.optional),
+
+  // Cell language (for smart-cells support)
+  language: CellLanguage.pipe(Schema.optional),
+
+  // Language-specific metadata (e.g., SQL engine, output flag)
+  languageMetadata: Schema.Record({
```

Comment:
You may want to scope this per language so switching doesn't lose or override the metadata. but that can be a future thing